pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -732,6 +736,28 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, _ v1.PodStatus, podStat
 			continue
 		}
 
+		// limit pod restart count if set
+		restartCountLimitStr, ok := pod.Annotations[podRestartCountLimitAnn]
+		if ok {
+			if len(restartCountLimitStr) > 0 {
+				restartCountLimit, err := strconv.Atoi(restartCountLimitStr)
+				if err != nil {
+					glog.V(5).Infof("Failed to parse restart count limit in pod %v: %+v", format.Pod(pod), err)
```

Comment:
would it be better to put it into the event pool of pod? then I can describe pod to see what happens to this pod.